ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof.nr

```diff
@@ -23,6 +25,45 @@ pub fn leaf_get_len<MAX_LEN>(leaf: Leaf<MAX_LEN>) -> u32 {
     leaf_unpadded_len as u32
 }
 
+// Prefixes mean:
+// 0 - Extension node, even number of nibbles
+// 1 - Extension node, odd number of nibbles
+// 2 - Leaf node, even number of nibbles
+// 3 - Leaf node, odd number of nibbles
+pub fn parity(prefix: u8) -> u64 {
+    assert([0, 1, 2, 3].any(|x| x == prefix), "Prefix must be 0, 1, 2 or 3");
+    let mut result = -1;
+    if (prefix % 2 == 0) {
+        result = EVEN;
+    } else {
+        result = ODD;
+    }
+    assert((result == EVEN) | (result == ODD), "Invalid parity");
+
+    result
+}
+
+// Leaf and extension nodes are RLP lists with 2 elements: prefixed key nibbles (in bytes) and value.
+// This function removes the prefix and returns appropriate number of key nibbles according to the parity of the prefix (see function: parity).
+pub fn unprefixed_key_nibbles<MAX_KEY_LEN, MAX_KEY_NIBBLE_LEN>(prefixed_key_bytes: [u8; MAX_KEY_LEN]) -> [u8; MAX_KEY_NIBBLE_LEN] {
```

Comment:
Perhaps you meant `unprefix_key_nibbles` i.e. strip prefix?

Reply:
Ahhh, ok that only return number of nibbles? Than just `count_nibbles` or `calculate_nibbles_count` or `get_nibbles_len`

Reply:
Because I wanted to include information that I convert given bytes to nibbles

Reply:
I changed it to `strip_prefix`